src/hotspot/share/opto/compile.cpp

```diff
@@ -986,6 +990,11 @@ Compile::Compile( ciEnv* ci_env,
   Code_Gen();
 }
 
+Compile::~Compile() {
+  delete _print_inlining_stream;
+  delete _first_failure_details;
```

Comment:
Should we first check for `nullptr` before deleting here?

Reply:
delete, like free, accepts nullptr and does nothing. We delete/os::free null in many places.